src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaGoalPersistenceAdapter.kt

```diff
@@ -53,7 +55,8 @@ class JpaGoalPersistenceAdapter(
     val goalEntity = getGoalEntityByPrisonNumberAndGoalReference(prisonNumber, updatedGoalDto.reference)
     return if (goalEntity != null) {
       goalMapper.updateEntityFromDto(goalEntity, updatedGoalDto)
-      goalMapper.fromEntityToDomain(goalEntity)
+      val persistedEntity = goalRepository.saveAndFlush(goalEntity)
```

Comment:
Previously, without the saveAndFlush, the goal entity was updated and the JPA fields were correctly being set and updated .... but the flush (and therefore the JPA managed fields set) was being done at the transaction boundary. This meant that the domain goal that was returned (via the mapper) did not have the correct updatedBy fields because they had not been set by JPA at that point

Reply:
OK, will do 👍

Reply:
OK, done